File: FoodDelivery.DAL/EF/Entities/RestaurantOrderItem.cs
using System;

namespace FoodDelivery.DAL.EF.Entities
{
	public class RestaurantOrderItem
	{
		public Guid Id { get; set; }
		public Guid RestaurantOrderId { get; set; }
		public Guid OrderItemId { get; set; }
		public int Status { get; set; }
		public int? RequestedQuantity { get; set; }

		public virtual RestaurantOrder RestaurantOrder { get; set; }
		public virtual OrderItem OrderItem { get; set; }
	}
}

File: FoodDelivery.Web/src/app/models/restaurantOrder/RestaurantOrderItem.ts
import { OrderItemStatus } from './../enums/statuses/OrderItemStatus';
import { OrderItem } from '../order/OrderItem';
import { Guid } from 'guid-typescript';

export class RestaurantOrderItem{
    public id: Guid
    public status: OrderItemStatus
    public orderItem: OrderItem;
    public requestedQuantity?: number;
}